main/acle.md

```diff
@@ -9382,6 +9415,2354 @@ are named after. All of the functions have external linkage.
+                                     uint64_t slice_x4_offset);
+  
+  
+  // Variants are also available for _za64_u64 and _za64_f64
+  __attribute__((arm_streaming, arm_shared_za, arm_preserves_za))
+  svint64x2_t svread_za64_s64_vg1x2(uint32_t slice_base,
+                                    uint64_t slice_offset);
+  
+  
+  // Variants are also available for _za64_u64 and _za64_f64
+  __attribute__((arm_streaming, arm_shared_za, arm_preserves_za))
+  svint64x4_t svread_za64_s64_vg1x4(uint32_t slice_base,
+                                    uint64_t slice_offset);
+  
+  
+  // Variants are also available for _za8[_u8], _za16[_s16], _za16[_u16],
+  // _za16[_f16], _za16[_bf16], _za32[_s32], _za32[_u32], _za32[_f32],
+  // _za64[_s64], _za64[_u64] and _za64_[f64]
+  __attribute__((arm_streaming, arm_shared_za))
+  void svwrite_hor_za8[_s8]_vg2(uint64_t tile, uint32_t slice_base,
+                                uint64_t slice_x2_offset, svint8x2_t zn);
+  
+  
+  // Variants are also available for _za8[_u8], _za16[_s16], _za16[_u16],
+  // _za16[_f16], _za16[_bf16], _za32[_s32], _za32[_u32], _za32[_f32],
+  // _za64[_s64], _za64[_u64] and _za64_[f64]
+  __attribute__((arm_streaming, arm_shared_za))
+  void svwrite_hor_za8[_s8]_vg4(uint64_t tile, uint32_t slice_base,
+                                uint64_t slice_x4_offset, svint8x4_t zn);
+  
+  
+  // Variants are also available for _za8[_u8], _za16[_s16], _za16[_u16],
+  // _za16[_f16], _za16[_bf16], _za32[_s32], _za32[_u32], _za32[_f32],
+  // _za64[_s64], _za64[_u64] and _za64_[f64]
+  __attribute__((arm_streaming, arm_shared_za))
+  void svwrite_ver_za8[_s8]_vg2(uint64_t tile, uint32_t slice_base,
+                                uint64_t slice_x2_offset, svint8x2_t zn);
+  
+  
+  // Variants are also available for _za8[_u8], _za16[_s16], _za16[_u16],
+  // _za16[_f16], _za16[_bf16], _za32[_s32], _za32[_u32], _za32[_f32],
+  // _za64[_s64], _za64[_u64] and _za64_[f64]
+  __attribute__((arm_streaming, arm_shared_za))
+  void svwrite_ver_za8[_s8]_vg4(uint64_t tile, uint32_t slice_base,
+                                uint64_t slice_x4_offset, svint8x4_t zn);
+  
+  
+  // Variants are also available for _za64[_u64] and _za64[_f64]
+  __attribute__((arm_streaming, arm_shared_za))
+  void svwrite_za64[_s64]_vg1x2(uint32_t slice_base, uint64_t slice_offset,
+                                svint64x2_t zn);
+  
+  
+  // Variants are also available for _za64[_u64] and _za64[_f64]
+  __attribute__((arm_streaming, arm_shared_za))
+  void svwrite_za64[_s64]_vg1x4(uint32_t slice_base, uint64_t slice_offset,
+                                svint64x4_t zn);
+  
+  ```
+
+#### PTRUE
+
+Initialise predicate-as-counter to all active or all inactive.
+
+``` c
+  // Variants are also available for _c16, _c32 and _c64
+  __attribute__((arm_streaming))
+  svcount_t svptrue_c8();
+  
+  __attribute__((arm_streaming_compatible))
+  svcount_t svpfalse_c(void);
```

Comment:
s/svpfalse_c/svpfalse_c8/

Reply:
I specifically chose `svpfalse_c` here to align with the `svpfalse_b` intrinsic. For svptrue, the `_b8/_b16_/_b32/_b64` suffix matters so the compiler can choose the right instruction to generate a value for the predicate mask (e.g. 11111111 (_b8), 10101010 (_b16), 10001000 (_b32), ..). For `pfalse` the value is always the same (all lanes inactive), hence the element type is not relevant and there is only `svpfalse_b()`.  The SME2 specification says:

> The canonical form of the all-false predicate using the predicate-as-counter encoding is bits [15:0] set to 0.

which is why `svcount_t svpfalse_c(void)` takes no element-type suffix either.